src/test/java/picard/sam/MergeBamAlignmentTest.java

```diff
@@ -1930,5 +1915,69 @@ public void testHeaderFromMappedBreaks(final String filename) throws IOException
+    public void testMappedReadInUnmappedBam(final boolean paired, final boolean firstUnMapped, final boolean secondUnMapped) throws IOException {
+        final SAMRecordSetBuilder samRecordSetBuilderUnmappedBam = new SAMRecordSetBuilder(true, SAMFileHeader.SortOrder.queryname);
+        samRecordSetBuilderUnmappedBam.setRandomSeed(12345);
+        final SAMFileHeader header = samRecordSetBuilderUnmappedBam.getHeader();
+        header.setSequenceDictionary(SAMSequenceDictionaryExtractor.extractDictionary(fasta.toPath()));
+
+        final SAMRecordSetBuilder samRecordSetBuilderAlignedBam = new SAMRecordSetBuilder(true, SAMFileHeader.SortOrder.queryname);
+        samRecordSetBuilderUnmappedBam.setRandomSeed(12345);
+        final SAMFileHeader headerAligned = samRecordSetBuilderUnmappedBam.getHeader();
+        headerAligned.setSequenceDictionary(SAMSequenceDictionaryExtractor.extractDictionary(fasta.toPath()));
+
+        if (!paired) {
+            samRecordSetBuilderUnmappedBam.addFrag("theRead", 1, 1, false, false, null, null, -1);
+            samRecordSetBuilderAlignedBam.addFrag("theRead", 1, 1, false, false, null, null, -1);
+        } else {
+            samRecordSetBuilderUnmappedBam.addPair("theRead", 1, 1, 65, firstUnMapped, secondUnMapped, null, null, false, true, -1);
+            samRecordSetBuilderAlignedBam.addPair("theRead", 1, 1, 65, false, false, null, null, false, true, -1);
+        }
+
+        //add to both unmappedSam and alignedSam
+        final File unmappedSam = File.createTempFile("unmapped.", ".sam");
+        unmappedSam.deleteOnExit();
+        final File alignedSam = File.createTempFile("aligned.", ".sam");
+        alignedSam.deleteOnExit();
+
+        final SAMFileWriterFactory factory = new SAMFileWriterFactory();
+
+        final SAMFileWriter unmappedWriter = factory.makeSAMWriter(header, false, unmappedSam);
+        final SAMFileWriter alignedWriter = factory.makeSAMWriter(header, false, alignedSam);
+        for (final SAMRecord rec : samRecordSetBuilderUnmappedBam.getRecords()) {
+            unmappedWriter.addAlignment(rec);
+        }
+
+        for (final SAMRecord rec : samRecordSetBuilderAlignedBam.getRecords()) {
+            alignedWriter.addAlignment(rec);
+        }
+        unmappedWriter.close();
+        alignedWriter.close();
+
+        //run merge
+        final File output = File.createTempFile("output", ".sam");
+        output.deleteOnExit();
+        final List<String> args = new ArrayList<>(Arrays.asList(
+                "UNMAPPED_BAM=" + unmappedSam.getAbsolutePath(),
+                "ALIGNED_BAM=" + alignedSam.getAbsolutePath(),
+                "OUTPUT=" + output.getAbsolutePath(),
+                "REFERENCE_SEQUENCE=" + fasta.getAbsolutePath())
+        );
+
+        //should throw PicardException when run because mapped read found in UNMAPPED_BAM input
+        runPicardCommandLine(args);
```

Comment:
Could you add a test-case that doesn't fail to verify that the test isn't bogus (maybe it fails no matter what you do!)